### PR TITLE
Use RPM builder instead of redlinerpm

### PR DESCRIPTION
**Pull request by wendigo**
It has superior configuration abilities and is a little faster (30 vs 45s on average).

https://ctron.github.io/rpm-builder/rpm-mojo.html
https://github.com/ctron/rpm-builder

### CODE REVIEW

File: core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java [hunk 1]
```diff
@@ -152,6 +190,24 @@ private static void assertPathDeleted(GenericContainer<?> container, String path
...
 
+    private static Map<String, FileInfo> getRpmFiles(String lines)
+    {
+        Splitter splitter = Splitter.onPattern("\s+").trimResults();
+        ImmutableMap.Builder<String, FileInfo> builder = ImmutableMap.builder();
+        for (String line : Splitter.on("\n").split(lines.trim())) {
+            List<String> columns = splitter.splitToList(line);
+            boolean isLink = columns.get(0).startsWith("l");
+
+            builder.put(columns.get(8), new FileInfo(
```
Comment (nineinchnick) [inline core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java:201]: nit: assert the number of items in `columns`, document the expected output of `rpm -qlpv`, and use named variables. Maybe something like:
```
// expect lines to look like:
// drwxr-xr-x    2 root    root     0 Apr  7  2021 /some/file
String permissions = columns.get(0);
String owner = columns.get(2);
String group = columns.get(3);
String path = columns.get(8);
String link = columns.get(10);
new FileInfo(path, PosixFilePermissions.fromString(permissions.substring(1)), owner, group, link);
```

or do that in an alternative `FileInfo` constructor.

**Review by nineinchnick — APPROVED** [review]

File: core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java (hunk 1 above)
Comment (wendigo) [reply core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java:201]: Yeah. I'll actually create more versatile assertions for these

Comment (nineinchnick) [reply core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java:201]: Did you push any changes here? It looks the same.

Comment (wendigo) [reply core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java:201]: Not yet

Comment (wendigo) [reply core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java]: it is now, ptal

**Comment by wendigo** [comment]
I've added a bunch of tests to assert on both metadata and RPM content

**Review by nineinchnick — APPROVED** [review]

File: core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java [hunk 2]
```diff
@@ -102,6 +112,67 @@ private void testInstall(String javaVersion)
         }
     }
 
+    @Test
+    public void testRpmContents()
```
Comment (electrum) [inline core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java:116]: We need to verify the symlinks to `shared` for JARs

File: core/trino-server-rpm/pom.xml [hunk 3]
```diff
@@ -11,11 +11,16 @@
 
     <artifactId>trino-server-rpm</artifactId>
     <packaging>rpm</packaging>
+    <description>Trino Server RPM package</description>
+
+    <!-- will be used by the rpm-builder -->
+    <url>https://trino.io</url>
```
Comment (electrum) [inline core/trino-server-rpm/pom.xml:17]: Shouldn't this be inherited from the parent POM?

File: core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java [hunk 4]
```diff
@@ -165,11 +165,11 @@ public void testRpmMetadata()
             assertThat(rpmMetadata).extractingByKey("Name").isEqualTo("trino-server-rpm");
             assertThat(rpmMetadata).extractingByKey("Epoch").isEqualTo("0");
             assertThat(rpmMetadata).extractingByKey("Release").isEqualTo("1");
-            assertThat(rpmMetadata).extractingByKey("Version").isEqualTo("440.SNAPSHOT"); // TODO: fixme
+            assertThat(rpmMetadata).extractingByKey("Version").isEqualTo("440-SNAPSHOT");
```
Comment (electrum) [inline core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java:168]: This needs to come from a property or something

File: core/trino-server-rpm/src/test/java/io/trino/server/rpm/TestDummy.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server.rpm;
+
+import org.junit.jupiter.api.Test;
+
+public class TestDummy
```
Comment (electrum) [inline core/trino-server-rpm/src/test/java/io/trino/server/rpm/TestDummy.java:18]: Why do we need to add this? It wasn't needed before

**Review by electrum — APPROVED** [review]

**Comment by electrum** [comment]
This is great! I'm happy we can deprecate our fork of the RPM plugin

File: core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java (hunk 2 above)
Comment (wendigo) [reply core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java:116]: Done.

File: core/trino-server-rpm/pom.xml (hunk 3 above)
Comment (wendigo) [reply core/trino-server-rpm/pom.xml:17]: It seems that it's not

File: core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java (hunk 4 above)
Comment (wendigo) [reply core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java:168]: It does come from the property.

File: core/trino-server-rpm/src/test/java/io/trino/server/rpm/TestDummy.java (hunk 5 above)
Comment (wendigo) [reply core/trino-server-rpm/src/test/java/io/trino/server/rpm/TestDummy.java:18]: I guess that the previous plugin was providing a test:test goal.

File: core/trino-server-rpm/pom.xml (hunk 3 above)
Comment (wendigo) [reply core/trino-server-rpm/pom.xml:17]: That's actually interesting. If I ran on master:

```
./mvnw help:effective-pom -pl ':trino-server' | grep -i url
Running `/Users/mateuszgajewski/Projects/src/github.com/trinodb/trino/mvnw`...
  <url>https://trino.io/core/trino-server</url>
```

Comment (wendigo) [reply core/trino-server-rpm/pom.xml:17]: So it's preexisting condition

Comment (wendigo) [reply core/trino-server-rpm/pom.xml:17]: According to the docs from maven xsd:

```
The URL to the project's homepage.
            <br><b>Default value is</b>: parent value [+ path adjustment] + (artifactId or project.directory property), or just parent value if
            project's <code>child.project.url.inherit.append.path="false"</code>
```

Comment (wendigo) [reply core/trino-server-rpm/pom.xml:17]: https://github.com/trinodb/trino/pull/20750

**Pull request merged**